### PR TITLE
feat(OpenVPN) renew server-side certificate

### PR DESCRIPTION
**Pull request by dduportal**
This PR aims at fixing https://github.com/jenkins-infra/helpdesk/issues/2798.

It introduces the following changes:

- Renewing the server-side certificate for OpenVPN (ref. https://community.openvpn.net/openvpn/wiki/EasyRSA3-OpenVPN-Howto for the minimalistic way of doing it, even though we did it with tooling from https://github.com/jenkins-infra/docker-openvpn)
- Moving all the OpenVPN "server" specific hieradata into client-level (instead of "common") to ensure that future "multi-servers" for OpenVPN could work + clarity 
  - Please note that the Docker image tag is kept in commons so both vagrant and production can use it

### CODE REVIEW

**Comment by dduportal** [comment]
Tested in production: the certificate + key had been already manually deployed for validation. This PR only allows the config as code to operate and overide manual changes

File: hieradata/clients/vpn.jenkins.io.yaml [hunk 1]
```diff
@@ -9,3 +9,13 @@ profile::openvpn::networks:
...
+# DH comes from https://github.com/jenkins-infra/docker-openvpn
```
Comment (lemeurherve) [inline hieradata/clients/vpn.jenkins.io.yaml:20]:
```suggestion
# DH comes from https://github.com/jenkins-infra/docker-openvpn, it should not have to be updated
```
Not sure if it should be added. WDYT?

File: hieradata/common.yaml [hunk 2]
```diff
@@ -338,5 +330,5 @@ limits:
     nofile:
       soft: '65536'
       hard: '65536'
-
+profile::openvpn::image_tag: '1.0.28' ## Tracked by updatecli
```
Comment (lemeurherve) [inline hieradata/common.yaml]:
```suggestion
profile::openvpn::image_tag: '1.0.28' # Tracked by updatecli
```
(nit)

**Review by lemeurherve — DISMISSED** [review]

File: hieradata/clients/vpn.jenkins.io.yaml (hunk 1 above)
Comment (dduportal) [reply hieradata/clients/vpn.jenkins.io.yaml:20]: Disagree on this suggestion: the comment implies to never update DH while there are cases where it is required.

**Review by lemeurherve — APPROVED** [review]

**Comment by dduportal** [comment]
Merging as admin: the CI error also appears on the production branch and is not caused by this change (but it has to be fixed)

**Pull request merged**